src/include/btree_inline.h

```diff
@@ -1485,21 +1485,16 @@ __wt_page_del_active(WT_SESSION_IMPL *session, WT_REF *ref, bool visible_all)
 }
 
 /*
- * __wt_btree_can_evict_dirty --
- *     Check whether eviction of dirty pages or splits are permitted in the current tree. We cannot
- *     evict dirty pages or split while a checkpoint is in progress, unless the checkpoint thread is
- *     doing the work. Also, during connection close, if we take a checkpoint as of a timestamp,
- *     eviction should not write dirty pages to avoid updates newer than the checkpoint timestamp
- *     leaking to disk.
+ * __wt_btree_in_checkpoint --
+ * Returns true if the session is performing a sync on its current tree.
  */
 static inline bool
-__wt_btree_can_evict_dirty(WT_SESSION_IMPL *session)
+__wt_btree_in_checkpoint(WT_SESSION_IMPL *session)
 {
     WT_BTREE *btree;
 
     btree = S2BT(session);
-    return ((!WT_BTREE_SYNCING(btree) || WT_SESSION_BTREE_SYNC(session)) &&
-      !F_ISSET(S2C(session), WT_CONN_CLOSING_TIMESTAMP));
+    return (!WT_BTREE_SYNCING(btree) || WT_SESSION_BTREE_SYNC(session));
```

Comment:
Is this flag used anywhere else WT_CONN_CLOSING_TIMESTAMP? Can we just remove that flag?

Reply:
It doesn't look like we can remove `WT_CONN_CLOSING_TIMESTAMP`, it is used in a few other places.